### PR TITLE
Update mafia.js

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

File: chat-plugins/mafia.js [hunk 1]
```diff
@@ -355,7 +355,7 @@ class MafiaTracker extends Rooms.RoomGame {
 		if (!player && this.dead[user.userid] && this.dead[user.userid].restless) player = this.dead[user.userid];
 		if (!(target in this.players) && target !== 'nolynch') return false;
 		if (player.lynching || (target === player.userid && !this.selfEnabled)) return false;
-		if (target === player.userid && (this.lynches[target] && this.getHammer() - 1 > this.lynches[target].count) && this.selfEnabled === 'hammer') return false;
+		if (target === player.userid && (this.lynches[target] && this.hammerCount - 1 > this.lynches[target].count) && this.selfEnabled === 'hammer') return false;
```
Comment (ZestOfLife) [inline chat-plugins/mafia.js]: Change this to `if (target === player.userid && (this.hammerCount - 1 > (this.lynches[target] ? this.lynches[target].count : 0)) && this.selfEnabled === 'hammer') return false;` (it fixes ppl lynching themselves when lynch obj doesnt exist of theirs)

**Comment by HoeenCoder** [comment]
Hey @whalemer could you be more descriptive in your commit messages in the future? It helps clarify what is being changed in a commit. https://github.com/Zarel/Pokemon-Showdown/blob/master/CONTRIBUTING.md#commit-standards

File: chat-plugins/mafia.js [hunk 2]
```diff
@@ -495,16 +501,36 @@ class MafiaTracker extends Rooms.RoomGame {
...
+			if (deadPlayer.treestump) delete deadPlayer.treestump;
+			if (deadPlayer.restless) delete deadPlayer.restless;
+			this.sendRoom(`${deadPlayer.name} was revived!`, {declare: true});
+			this.players[deadPlayer.userid] = deadPlayer;
+			this.roles.push(deadPlayer.role);
+			delete this.dead[deadPlayer.userid];
+		} else {
+			const targetUser = Users(deadPlayer);
+			if (!targetUser || !targetUser.connected) return user.sendTo(this.room, `|error|The user "${deadPlayer}" was not found.`);
+			if (!this.room.users[targetUser.userid]) return this.errorReply(`${targetUser.name} is not in this room, and cannot be hosted.`);
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: added not hosted

File: chat-plugins/mafia.js [hunk 3]
```diff
@@ -495,16 +501,36 @@ class MafiaTracker extends Rooms.RoomGame {
...
+			let player = this.makePlayer(targetUser);
+			if (this.started) {
+				player.lynching = '';
+				player.role = {
+					name: `No set role`,
+					id: `nosetrole`,
+					alignment: 'solo',
+					memo: [`You were added to the game after it had started. To learn about your role, PM the host (${this.host}).`],
+				};
+				this.roles.push(player.role);
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: update the role string too

File: chat-plugins/mafia.js [hunk 4]
```diff
@@ -495,16 +501,36 @@ class MafiaTracker extends Rooms.RoomGame {
...
+			} else {
+				this.roles = [];
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: reset the other role values too

File: chat-plugins/mafia.js [hunk 5]
```diff
@@ -495,16 +501,36 @@ class MafiaTracker extends Rooms.RoomGame {
...
+			delete this.dead[deadPlayer.userid];
+		} else {
+			const targetUser = Users(deadPlayer);
+			if (!targetUser || !targetUser.connected) return user.sendTo(this.room, `|error|The user "${deadPlayer}" was not found.`);
+			if (!this.room.users[targetUser.userid]) return this.errorReply(`${targetUser.name} is not in this room, and cannot be hosted.`);
+			let player = this.makePlayer(targetUser);
+			if (this.started) {
+				player.lynching = '';
+				player.role = {
+					name: `No set role`,
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: Id rather call it "Unknown"

File: chat-plugins/mafia.js [hunk 6]
```diff
@@ -495,16 +501,36 @@ class MafiaTracker extends Rooms.RoomGame {
...
+			this.players[deadPlayer.userid] = deadPlayer;
+			this.roles.push(deadPlayer.role);
+			delete this.dead[deadPlayer.userid];
+		} else {
+			const targetUser = Users(deadPlayer);
+			if (!targetUser || !targetUser.connected) return user.sendTo(this.room, `|error|The user "${deadPlayer}" was not found.`);
+			if (!this.room.users[targetUser.userid]) return this.errorReply(`${targetUser.name} is not in this room, and cannot be hosted.`);
+			let player = this.makePlayer(targetUser);
+			if (this.started) {
+				player.lynching = '';
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: No need to reset this, it will be `''` since you just made the player.

File: chat-plugins/mafia.js [hunk 7]
```diff
@@ -654,6 +669,7 @@ class MafiaTracker extends Rooms.RoomGame {
 	}
 
 	forfeit(user) {
+		if (!(user.userid in this.players)) return user.sendTo(this.room, `|error|You are not a player in the game.`);
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: feel free to just return false here.

File: chat-plugins/mafia.js (hunk 3 above)
Comment (whalemer) [reply chat-plugins/mafia.js]: that part doesnt return, the role string, playercount etc are done at 533

**Pull request merged**